tensorboard/webapp/runs/store/runs_reducers_test.ts

```diff
@@ -831,4 +833,143 @@ describe('runs_reducers', () => {
       expect(nextState.data.runColorOverrideForGroupBy).toEqual(new Map());
     });
   });
+
+  describe('#stateRehydratedFromUrl', () => {
+    it('ignores non-dashboard routeKind', () => {
+      const state = buildRunsState({
+        initialGroupBy: {key: GroupByKey.EXPERIMENT},
+        userSetGroupBy: {key: GroupByKey.RUN},
+      });
+
+      const nextState = runsReducers.reducers(
+        state,
+        stateRehydratedFromUrl({
+          routeKind: RouteKind.EXPERIMENTS,
+          partialState: {},
+        })
+      );
+
+      expect(nextState.data.userSetGroupBy).toEqual({key: GroupByKey.RUN});
+    });
+
+    it('sets userSetGroupBy to the value provided', () => {
+      const state = buildRunsState({
+        initialGroupBy: {key: GroupByKey.REGEX, regexString: ''},
+        userSetGroupBy: {key: GroupByKey.RUN},
+      });
+
+      const partialState: URLDeserializedState = {
+        runs: {
+          groupBy: {key: GroupByKey.EXPERIMENT},
+        },
+      };
+      const nextState = runsReducers.reducers(
+        state,
+        stateRehydratedFromUrl({
+          routeKind: RouteKind.EXPERIMENT,
+          partialState,
+        })
+      );
+
+      expect(nextState.data.userSetGroupBy).toEqual({
+        key: GroupByKey.EXPERIMENT,
+      });
+    });
+
+    it('ignores the action if the partialState does not contain groupBy', () => {
+      const state = buildRunsState({
+        initialGroupBy: {key: GroupByKey.EXPERIMENT},
+        userSetGroupBy: {key: GroupByKey.RUN},
+      });
+
+      const partialState: URLDeserializedState = {
+        runs: {
+          groupBy: null,
+        },
+      };
+      const nextState = runsReducers.reducers(
+        state,
+        stateRehydratedFromUrl({
+          routeKind: RouteKind.EXPERIMENT,
+          partialState,
+        })
+      );
+
+      expect(nextState.data.userSetGroupBy).toEqual({
+        key: GroupByKey.RUN,
+      });
+    });
+
+    // Color grouping potentially is an expensive operation and assigning colors
+    // on route changes may not actually be effective at all. Because we are
+    // using RouteContextState, color assignment and groupBy information should
+    // not go out of sync. It would only be used to assign initial groupBy
+    // information from URL that will often trigger a run fetch which will be
+    // handled separately by the fetchRuns flow.
```

Comment:
iiuc, the gist of this comment is that stateRehydratedFromUrl intentionally does not update color assignment, because it is expensive, so we rely on the fetchRunsSucceeded to handle it.

Suggestion: can we put this comment in the runs_reducers's stateRehydratedFromUrl handler?

> assigning colors on route changes may not actually be effective at all
Could you please help me understand what this means?

> color assignment and groupBy information should not go out of sync
Could you elaborate on what this means?  In this PR, the reducer now updates groupBy without updating color assignments, which suggests that color assignment and groupBy do not need to be kept in sync.

nit: RouteContextState --> RouteContextedState

Reply:
Left a comment in the code instead of leaving the context on GitHub. PTAL.